File: src/main/java/com/example/backend/dto/tarot/TarotResponse.java
package com.example.backend.dto.tarot;

import com.example.backend.dto.common.Translatable;
import com.example.backend.lang.Language;
import java.util.List;

public record TarotResponse(
    List<TarotCardDto> cards,
    String text,
    Language from,
    Language to) implements Translatable {


    public TarotResponse {
    }

    public TarotResponse(List<TarotCardDto> cards, String text) {
        this(cards, text, Language.EN, Language.EN);
    }
}

File: src/main/kotlin/com/example/backend/dto/tarot/TarotResponse.kt
package com.example.backend.dto.tarot

import com.example.backend.dto.common.Translatable
import com.example.backend.lang.Language

data class TarotResponse(
    var cards: List<TarotCardDto>? = null,
    var text: String? = null,
    var from: Language? = Language.EN,
    var to: Language? = Language.EN
) : Translatable {

    override fun from(): Language? {
        return from
    }

    override fun to(): Language? {
        return to
    }

    override fun text(): String? {
        return text
    }
}
